parachain/pallets/inbound-queue/src/lib.rs

```diff
@@ -214,20 +223,23 @@ pub mod pallet {
 
 			// Decode message into XCM
 			let xcm = match inbound::VersionedMessage::decode_all(&mut envelope.payload.as_ref()) {
-				Ok(inbound::VersionedMessage::V1(message_v1)) => message_v1.into(),
+				Ok(message) => inbound::VersionedMessageToXcmConverter::<
```

Comment:
A couple of issues with this implementation:

1. The inbound queue does not need to know about  the `RegisterCallIndex` type. That is an implementation detail of the converter. I would remove it from the pallet config.

2. Instead, the converter itself should be provided to the pallet config. Example:

```rust
type MessageConverter: ConvertMessage
```

where `ConvertMessage` is a trait implemented by the converter:

```
pub trait ConvertMessage {
  fn convert(message: VersionedMessage) -> Result<Xcm<()>, ConvertMessageError>;
}

pub enum ConvertMessageError {
  UnsupportedVersion,
}
```

Reply:
Thanks for the suggestion! Some revamp in https://github.com/Snowfork/snowbridge/pull/961/commits/18a90eb78006ebbd441184d89dcb9fdc9535e3a2#diff-7d758f1d757c44217a6d64358b4e11fa6fadaf9d160c655ee9b076257dd1be86